content/blog/2018-10-23-react-v-16-6.md

```diff
@@ -0,0 +1,149 @@
+
+```js
+const MyComponent = React.memo(function MyComponent(props) {
+  /* only rerenders if props change */
+});
+```
+
+## [`React.lazy`](/docs/code-splitting.md#reactlazy) - Code-Splitting with `Suspense`
+
+You may have seen [Dan's talk about React Suspense at JSConf Iceland](/blog/2018/03/01/sneak-peek-beyond-react-16.html). Now you can use the Suspense component to do [code-splitting](/docs/code-splitting.md#reactlazy) by wrapping a dynamic import in a call to `React.lazy()`.
+
+```js
+import React, {lazy, Suspense} from 'react';
+const OtherComponent = lazy(() => import('./OtherComponent'));
+const MyComponent = () => (
+  <Suspense fallback={<div>Loading...</div>}>
+    <OtherComponent />
+  </Suspense>
+);
+```
+
+The Suspense component will also allow library authors to start building data fetching with Suspense support.
+
+> Note: This feature is not yet available for server-side rendering. Suspense support will be added in a later release.
+
+## [`static contextType`](/docs/context.md#classcontexttype)
+
+In [React 16.3](/blog/2018/03/29/react-v-16-3.html) we introduced the official Context API as a replacement to the previous [Legacy Context](/docs/legacy-context.html) API.
+
+```js
+const MyContext = React.createContext();
+```
+
+We've heard feedback that adopting the new render prop API can be difficult in class components. So we've add a convenience API to [consume a context value from within a class component](/docs/context.md#classcontexttype).
+
+```js
+class MyClass extends React.Component {
+  static contextType = MyContext;
+  componentDidMount() {
+    let value = this.context;
+    /* perform a side-effect at mount using the value of MyContext */
+  }
+  componentDidUpdate() {
+    let value = this.context;
+    /* ... */
+  }
+  componentWillUnmount() {
+    let value = this.context;
+    /* ... */
+  }
+  render() {
+    let value = this.context;
+    /* render something based on the value of MyContext */
+  }
+}
+```
+
+## [`static getDerivedStateFromError()`](/docs/react-component.html#static-getderivedstatefromerror)
+
+React 16 introduced [Error boundaries](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html) for handling errors thrown in React renders. We already had the `componentDidCatch` life-cycle which gets fired after an error has already happened. It's great for logging errors to the server. It also lets you show a different UI to the user by calling `setState`.
+
+Before that is fired, we render `null` in place. This can be confusing to parent components that now have `null` in their refs. It also doesn't work to recover from errors on the server since the `Did` life-cycles doesn't fire during server-side rendering.
```

Comment:
Nit: "lifecycle methods"
Nit: "don't fire"

I think people won't understand what "`null` in their refs" means. Maybe "This sometimes break parent components that don't expect their refs to be empty."